src/sdam/topology.ts

```diff
@@ -903,10 +903,21 @@ function processWaitQueue(topology: Topology) {
       continue;
     } else if (selectedDescriptions.length === 1) {
       selectedServer = topology.s.servers.get(selectedDescriptions[0].address);
+    } else if (selectedDescriptions.length === 2) {
```

Comment:
Why not fall through to the `else` here instead of special-casing two servers?   If it's just to get a slight performance boost, it doesn't seem worth it unless it's measurable and significant.

Reply:
Micro benchmarks is all I got so we can go with the assumption that the performance will be outclassed by other steps in the process of executing an operation and just use shuffle here. We discussed optimizing the 2 item case within shuffle, which we can do separately, and that'll be best of both worlds

Reply:
```txt
shuffle(["a", "b"], 2)                x 36,828,427 ops/sec ±0.25% (98 runs sampled)
Math.random() > 0.5 ? item1 : item2   x 155,421,810 ops/sec ±0.10% (98 runs sampled)
Math.floor(Math.random() * 2) === 0   x 135,506,163 ops/sec ±0.12% (101 runs sampled)

Fastest is Math.random() > 0.5 ? item1 : item2
```

Reply:
Since an application isn't likely to break a few thousand operations per second, this is unlikely to make a difference in practice.